### PR TITLE
Fix for FlipView : Navigation's button visibility and Transition update

### PR DESCRIPTION
**Pull request by jumulr**
Refresh navigation buttons' visibility when control's orientation changes.
Handle navigation buttons' visibility when toggling their display.
Apply transition when SelectedIndex changes.

### CODE REVIEW

File: src/MahApps.Metro/MahApps.Metro.Shared/Controls/FlipView.cs [hunk 1]
```diff
@@ -49,19 +51,62 @@ public class FlipView : Selector
...
+                CoerceValueCallback = (d, value) =>
+                {
+                    /* Chain coercions. */
+                    if (!Object.Equals(previousSelectedIndexPropertyMetadata.CoerceValueCallback, null))
+                        foreach (var item in previousSelectedIndexPropertyMetadata.CoerceValueCallback.GetInvocationList())
+                            value = ((CoerceValueCallback)item)(d, value);
+                    return CoerceSelectedIndexProperty(d, value);
+                }
+            });
         }
```
Comment (punker76) [inline src/MahApps.Metro/MahApps.Metro.Shared/Controls/FlipView.cs]: Can you add more documentation what this code does?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/MahApps.Metro.Demo.Shared/ExampleViews/OtherExamples.xaml [hunk 2]
```diff
@@ -171,18 +181,42 @@
...
                                ItemTemplate="{StaticResource ImageDataTemplate}"
                                ItemsSource="{Binding FlipViewImages, Mode=OneWay}" />
+            <StackPanel Orientation="Horizontal" DataContext="{Binding ElementName=FlipView2nd, Mode=OneWay}" HorizontalAlignment="Center">
+                <Controls:NumericUpDown Controls:TextBoxHelper.SelectAllOnFocus="True"
+                                        ButtonsAlignment="Left"
+                                        Interval="1"
+                                        Minimum="1"
+                                        Maximum="{Binding Path=Items.Count, Mode=OneWay}"
+                                        Value="{Binding Path=SelectedIndex, Mode=TwoWay, Converter={vc:Int32IndexToNumberConverter}}"/>
+                <TextBlock VerticalAlignment="Center" Margin="8 0 0 0">
```
Comment (punker76) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/MahApps.Metro.Demo.Shared/ExampleViews/OtherExamples.xaml]: Why not starting from 0? Then you don't need the Int32IndexToNumberConverter.

**Comment by jumulr** [comment]
Here you are, i added a detail explanation of the coercion trick.
As i am totally new to GitHub, feel free to instruct me on the workflow in case i did a mistake.

**Comment by punker76** [comment]
@jumulr why closed?

**Comment by jumulr** [comment]
Mistake, sorry. Is that ok ?

**Pull request merged**